### PR TITLE
CI: Fix cuda toolkit speed issue.

### PR DESCRIPTION
**Pull request by wkpark**
Currently, CI uses the [Jimver/cuda-toolkit](https://github.com/Jimver/cuda-toolkit) for Windows which has known speed issues when working on Windows.
https://github.com/Jimver/cuda-toolkit/issues/253

specific package `"visual_studio_integration"` is the cause of this speed issue but it is needed to use msbuild.

here is the solution.
- do not install `"visual_studio_integration"` sub-package
- compile using MSVC + ninja, instead of using msbuild.
- and also turn off `github-cache` for cuda-tools to fix some broken cases.
<details><summary>Cache error log for example.</summary>

https://github.com/TimDettmers/bitsandbytes/actions/runs/7804268267/job/21285853144
~~~bash
Run Jimver/cuda-toolkit@v0.[2](https://github.com/TimDettmers/bitsandbytes/actions/runs/7804268267/job/21285853144#step:5:2).14
  with:
    cuda: 12.1.0
    method: local
    sub-packages: []
    non-cuda-sub-packages: []
    linux-local-args: ["--toolkit", "--samples"]
    use-github-cache: true
    use-local-cache: true
    log-file-suffix: log.txt
"C:\Program Files\Git\usr\bin\tar.exe" --posix -cf cache.tzst --exclude cache.tzst -P -C D:/a/bitsandbytes/bitsandbytes --files-from manifest.txt --force-local --use-compress-program "zstd -T0"
Cache Size: ~[3](https://github.com/TimDettmers/bitsandbytes/actions/runs/7804268267/job/21285853144#step:5:3)195 MB (3350[4](https://github.com/TimDettmers/bitsandbytes/actions/runs/7804268267/job/21285853144#step:5:4)07[6](https://github.com/TimDettmers/bitsandbytes/actions/runs/7804268267/job/21285853144#step:5:6)[7](https://github.com/TimDettmers/bitsandbytes/actions/runs/7804268267/job/21285853144#step:5:7)4 B)
Cache saved successfully
Error: Error: Got no files in tool cahce
~~~

</details>

without this fix: total build time is about ~30 ~40min
with this fix: total build time will be reduced about ~25min

### CODE REVIEW

File: .github/workflows/python-package.yml
Comment (rickardp) [inline .github/workflows/python-package.yml:106]: Maybe it is quicker to also do a network install here to avoid downloading the entire installer now that it's not cached?

File: .github/workflows/python-package.yml
Comment (rickardp) [inline .github/workflows/python-package.yml:130]: Cmake is already installed above

File: .github/workflows/python-package.yml
Comment (rickardp) [inline .github/workflows/python-package.yml:64]: Should this be only in the windows case?

**Review by rickardp — COMMENTED** [review]
Nice spot. I did not realize msvc worked without the VS integration of CUDA.

File: .github/workflows/python-package.yml
Comment (wkpark) [reply .github/workflows/python-package.yml:64]: Both work fine.

File: .github/workflows/python-package.yml
Comment (wkpark) [reply .github/workflows/python-package.yml:106]: I guess, the local cache of the `cuda-toolkit*exe` appears to be causing some issues due to its large size. maybe `network` will work with `use-github-cache: true`,

File: .github/workflows/python-package.yml
Comment (rickardp) [reply .github/workflows/python-package.yml:64]: Yes, --config seems to be a no-op on Linux

File: .github/workflows/python-package.yml
Comment (wkpark) [reply .github/workflows/python-package.yml:106]: it can get more speed with `'network'`. thanks!

`use-github-cache: true` also not work with `'network'`. it seems like a bug of `Jimver/cuda-toolkit`.

**Review by younesbelkada — APPROVED** [review]
Thanks for the clean fix !

**Comment by Titus-von-Koeller** [comment]
Thanks @wkpark for the clean fix and @rickardp for the review. Looks good also to me!

Thanks all for the intervention last week to get everything working again after the premature merge. Shouldn't have done that so hastily, it was based on a miscommunication.

I also sent you both an invite for the bitsandbytes Slack channel. In case you want to coordinate PRs, discuss sth important or I don't see something important in time which would make sense to unblock, please feel free to message there or contact me directly through Slack PM.

**Pull request merged**
